src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -168,6 +168,37 @@ private async Task StartDashboardAsync(CancellationToken cancellationToken = def
         };
 
         await kubernetesService.CreateAsync(dashboardExecutable, cancellationToken).ConfigureAwait(false);
+
+        await WaitForHttpSuccessOrThrow(dashboardUrl, TimeSpan.FromSeconds(DashboardWaitTimeInSeconds), cancellationToken).ConfigureAwait(false);
+    }
+
+    private const int DashboardWaitTimeInSeconds = 10;
+
+    private async Task WaitForHttpSuccessOrThrow(string url, TimeSpan timeout, CancellationToken cancellationToken = default)
+    {
+        using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+        cts.CancelAfter(timeout);
+
+        var client = new HttpClient();
+
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            try
+            {
+                var response = await client.GetAsync(url, cancellationToken).ConfigureAwait(false);
```

Comment:
Consider pulling down headers only:

```suggestion
                var response = await client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
```

Or just sending a HEAD request:

```suggestion
                var response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Head, url)).ConfigureAwait(false);
```